### PR TITLE
implement pool management commands

### PR DESCRIPTION
**Pull request by harshavardhana**
```
λ mc admin decommission start alias/ http://minio{1...2}/data{1...4}
```

```
λ mc admin decommission status alias/
┌─────┬─────────────────────────────────┬──────────────────────────────────┬────────┐
│ ID  │ Pools                           │ Capacity                         │ Status │
│ 1st │ http://minio{1...2}/data{1...4} │ 439 GiB (used) / 561 GiB (total) │ Active │
│ 2nd │ http://minio{3...4}/data{1...4} │ 329 GiB (used) / 421 GiB (total) │ Active │
└─────┴─────────────────────────────────┴──────────────────────────────────┴────────┘
```

```
λ mc admin decommission status alias/ http://minio{1...2}/data{1...4}
Progress: ===================> [1GiB/sec] [15%] [4TiB/50TiB]
Time Remaining: 4 hours (started 3 hours ago)
```

```
λ mc admin decommission status alias/ http://minio{1...2}/data{1...4}
ERROR: This pool is not scheduled for decommissioning currently.
```

```
λ mc admin decommission cancel alias/
┌─────┬─────────────────────────────────┬──────────────────────────────────┬──────────┐
│ ID  │ Pools                           │ Capacity                         │ Status   │
│ 1st │ http://minio{1...2}/data{1...4} │ 439 GiB (used) / 561 GiB (total) │ Draining │
└─────┴─────────────────────────────────┴──────────────────────────────────┴──────────┘
```

> NOTE: Canceled decommission will not make the pool active again, since we might have
> Potentially partial duplicate content on the other pools, to avoid this scenario be
> very sure to start decommissioning as a planned activity.

```
λ mc admin decommission cancel alias/ http://minio{1...2}/data{1...4}
┌─────┬─────────────────────────────────┬──────────────────────────────────┬────────────────────┐
│ ID  │ Pools                           │ Capacity                         │ Status             │
│ 1st │ http://minio{1...2}/data{1...4} │ 439 GiB (used) / 561 GiB (total) │ Draining(Canceled) │
└─────┴─────────────────────────────────┴──────────────────────────────────┴────────────────────┘
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -18,6 +18,7 @@ require (
 	github.com/minio/colorjson v1.0.1
 	github.com/minio/filepath v1.0.0
 	github.com/minio/madmin-go v1.2.2
+	github.com/minio/md5-simd v1.1.2 // indirect
```
Comment (klauspost) [inline go.mod:21]: Why did it do this??

File: cmd/admin-decom.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	Name:            "decommission",
+	Usage:           "manage MinIO server pool decommissioning",
+	Action:          mainAdminDecommission,
+	Before:          setGlobalsFromContext,
+	Flags:           globalFlags,
+	Subcommands:     adminDecommissionSubcommands,
+	HideHelpCommand: true,
+}
+
+// mainAdminDecommission is the handle for "mc admin config" command.
```
Comment (klauspost) [inline cmd/admin-decom.go:40]:
```suggestion
// mainAdminDecommission is the handle for "mc admin decommission" command.
```

File: cmd/admin-decom-cancel.go [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+	"github.com/fatih/color"
+	"github.com/minio/cli"
+	"github.com/minio/madmin-go"
+	"github.com/minio/mc/pkg/probe"
+	"github.com/minio/pkg/console"
+)
+
+var adminDecommissionCancelCmd = cli.Command{
+	Name:         "cancel",
+	Usage:        "cancel an ongoing decommissioning of a pool",
```
Comment (klauspost) [inline cmd/admin-decom-cancel.go:33]: Probably not the place, but I don't quite understand what it does.

File: cmd/admin-decom-cancel.go [hunk 4]
```diff
@@ -0,0 +1,136 @@
+// Copyright (c) 2015-2021 MinIO, Inc.
```
Comment (klauspost) [inline cmd/admin-decom-cancel.go:1]:
```suggestion
// Copyright (c) 2015-2022 MinIO, Inc.
```

:D

**Review by klauspost — APPROVED** [review]
Just some minors. LGTM.

File: go.mod (hunk 1 above)
Comment (harshavardhana) [reply go.mod:21]: No idea need to check

File: cmd/admin-decom-cancel.go [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+	CustomHelpTemplate: `NAME:
+  {{.HelpName}} - {{.Usage}}
+
+USAGE:
+  {{.HelpName}} TARGET
+
+FLAGS:
+  {{range .VisibleFlags}}{{.}}
+  {{end}}
+EXAMPLES:
```
Comment (krisis) [inline cmd/admin-decom-cancel.go:47]: Could we add an example to show how to cancel all ongoing decommissioning of pools?

**Review by krisis — APPROVED** [review]

File: cmd/admin-decom-cancel.go (hunk 5 above)
Comment (harshavardhana) [reply cmd/admin-decom-cancel.go:47]: Cancel is not recommended @krisis - not added on purpose.

Comment (harshavardhana) [reply cmd/admin-decom-cancel.go:47]: I will cancel example..

**Pull request merged**